evalml/pipelines/components/transformers/column_selectors.py

```diff
@@ -122,3 +123,44 @@ def transform(self, X, y=None):
             pd.DataFrame: Transformed X.
         """
         return super().transform(X, y)
+
+
+class SelectByTypeorTag(ColumnSelector):
+    """
+    Selects columns by specified Woodwork logical type or semantic tag in input data.
+
+    Arguments:
+        columns (list(string), list(ww.LogicalType)): List of datatypes, used to determine which columns to select.
+        random_seed (int): Seed for the random number generator. Defaults to 0.
+    """
+
+    name = "Select Columns By Type or Tag Transformer"
+    hyperparameter_ranges = {}
+    """{}"""
+    needs_fitting = False
+
+    def _check_input_for_columns(self, X):
+        """
+        This check is not required, since ww.select does not throw an error if the requested types are not in the dataset.
+        """
+        return
```

Comment:
Why define this method as a no-op? Why not simply omit it?

Reply:
The function is implemented in the parent class - I defined it here in order to overwrite the parent function.

Reply:
I see that `ColumnSelector._check_input_for_columns` does the following:
```python
        cols = self.parameters.get("columns") or []

        column_names = X.columns

        missing_cols = set(cols) - set(column_names)
        if missing_cols:
            raise ValueError(
                "Columns {} not found in input data".format(
                    ", ".join(f"'{col_name}'" for col_name in missing_cols)
                )
            )
```
And that the above is called during `ColumnSelector.fit`. And, since `SelectByType` doesn't define a `fit` method, `SelectByType.fit` will implicitly call `ColumnSelector.fit`, which is good 👍 

So, is there a reason why we should *not* also do that during `SelectByType.fit`?

I propose
* Make the change I suggested about defining an init method for `SelectByType` and accepting `column_types` as a separate arg. So the `columns` arg will have the same structure in both `ColumnSelector` and `SelectByType` init
* Delete `SelectByType._check_input_for_columns`